models/meshmodel/registry/registry.go

```diff
@@ -242,7 +243,14 @@ func (rm *RegistryManager) GetModels(db *database.Handler, f types.Filter) ([]v1
 		fmt.Println(modelWithCategoriess)
 		fmt.Println(err.Error()) //for debugging
 	}
-	for _, modelDB := range modelWithCategoriess {
+
+	for num, modelDB := range modelWithCategoriess {
+		filter := &v1alpha1.ComponentFilter{}
+		entities, _, _ := rm.GetEntities(filter)
+		host := rm.GetRegistrant(entities[num])
+		modelDB.HostID = host.ID
```

Comment:
Update host details after retrieving with GetModel function https://github.com/meshery/meshkit/pull/337/files#diff-2ff4deb53927cd3792edbed666384a10774880ce34a164439a027d0044dd6729R254
Split this append and GetModel logic and beflre appending to model update hostdetails.
Because we don’t want to change underlying model schema just want to add host details to user facing response.

Reply:
@MUzairS15, what about updating it the same way I'm retrieving hosts details for components and relationships directly in the handler

Reply:
Ok